### PR TITLE
Add Avatar to show image given userid

### PR DESCRIPTION
**Pull request by tomneedham**
Created a new component to show an avatar given the userid. 

 - loading spinner whilst we try to get it
 - avatar shown if available
 - placeholder icon if not

todo
 - [x] server url
 - [x] global avatar config?

### CODE REVIEW

File: src/components/OcAvatar.vue [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      /**
+       * Set to object URL when loaded, or on failure, icon placeholder is shown
+       */
+      avatarSource: '',
+      loading: true
+    }
+  },
+  mounted: function () {
+    let headers = new Headers()
+    let url = 'http://cloud.local/remote.php/dav/avatars/'+this.userid+'/128.png'
```
Comment (tomneedham) [inline src/components/OcAvatar.vue]: TODO

Comment (DeepDiver1975) [reply src/components/OcAvatar.vue]: fixed

File: src/components/Top-Bar.vue [hunk 2]
```diff
@@ -7,7 +7,7 @@
     </oc-topbar-item>
 
     <oc-topbar-item slot="right">
-      <oc-icon name="account_circle" color="white"></oc-icon>
+      <ocAvatar class="uk-margin-small-right" :userid="user.displayname" />
```
Comment (tomneedham) [inline src/components/Top-Bar.vue]: @DeepDiver1975 from what I found, we dont have the userid in the user object here - thoughts?

Comment (DeepDiver1975) [reply src/components/Top-Bar.vue]: should be user.id

Comment (DeepDiver1975) [reply src/components/Top-Bar.vue]: ah - missing in vuex

Screenshot from 2019-03-28 11-52-25

File: src/components/Avatar.vue [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      loading: true
+    }
+  },
+  mounted: function () {
+    if (!this.enabled) {
+      return;
+    }
+    let headers = new Headers()
+    let instance = this.$root.config.server || window.location.origin
+    let url = instance + '/remote.php/dav/avatars/' + this.userid + '/128.png'
```
Comment (DeepDiver1975) [inline src/components/Avatar.vue]: at time of mounting userid can not yet be set - better move this into a watcher on the property and load upon change

File: src/components/Top-Bar.vue (hunk 2 above)
Comment (DeepDiver1975) [reply src/components/Top-Bar.vue]: pushed a commit to add the is - now having troubles with the initial binding - see my comment on watching the prop .....

**Pull request closed without merging**